omod/src/main/java/org/openmrs/module/patientflags/web/FindFlaggedPatientsController.java

```diff
@@ -159,9 +159,21 @@ public ModelAndView processSubmit(@ModelAttribute("filter") Filter filter, Bindi
 		
 		// create the model map
 		ModelMap model = new ModelMap();
-		model.addAttribute("flaggedPatients", flaggedPatients.getMemberIds());
 		model.addAttribute("allPatients", allPatients);
-		
+		List<Map<String, Integer>> flaggedPatientList = new ArrayList<>();
+
+		Set<Integer> flaggedPatientIds = flaggedPatients.getMemberIds();
+		for (Integer patientId : flaggedPatientIds) {

```

Comment:
I think we could simplify this code using streams. Can you give it a try?

Reply:
I refactor the code using stream API